app/src/main/java/org/xbmc/kore/provider/MediaProvider.java

```diff
@@ -663,6 +668,16 @@ private SelectionBuilder buildQuerySelection(Uri uri, int match) {
                               .where(MediaContract.Songs.HOST_ID + "=?", hostId)
                               .where(MediaContract.Songs.ALBUMID + "=?", albumId);
             }
+            case SONGS_ARTIST: {
+                final String hostId = MediaContract.Hosts.getHostId(uri);
+                final String artistId = MediaContract.Artists.getArtistId(uri);
+                return builder.table(MediaDatabase.Tables.SONGS)
+                              .join(MediaDatabase.Tables.ALBUM_ARTISTS)
+                              .where(MediaDatabase.Tables.SONGS + "." + MediaContract.Songs.HOST_ID + "=?", hostId)
+                              .where(MediaDatabase.Tables.SONGS + "." + MediaContract.Songs.ALBUMID + "=" +
+                                     MediaDatabase.Tables.ALBUM_ARTISTS + "." + MediaContract.AlbumArtists.ALBUMID)
+                              .where(MediaContract.AlbumArtists.ARTISTID + "=?", artistId);
+            }
```

Comment:
This section doesn't follow the same pattern as the others. 
Take a look at the `ARTIST_ALBUMS_LIST` section for an example. You'll need to create a `MediaDatabase.Tables.SONGS_FOR_ARTIST_JOIN` (similar to `MediaDatabase.Tables.ALBUMS_FOR_ARTIST_JOIN`) that does the join and then use it in this section.
No need to introduce the `builder.join` method.